src/openpgp/packet/keymaterial.iced

```diff
@@ -461,15 +461,51 @@ class KeyMaterial extends Packet
 
   # Returns non-zero expire time if it exists, otherwise null.
   get_expire_time : () ->
-    return null unless (psc = @get_psc())?
-    list = psc.lookup[if @is_primary() then "self_sig" else "subkey_binding"]
+    if not (psc = @get_psc())? then null
+    else if @is_primary() then @_get_expire_time_on_primary()
+    else @_get_expire_time_on_subkey()
+
+  #-------------------
+
+  _get_expire_time_on_primary : () ->
+    table = @get_psc().lookup.self_sigs_by_uid
+
+    winner = null
+    key_generated = @timestamp
+
+    for uid,list of table
+
+      uid_winner = null
+      for packetsig in list when (sig = packetsig.sig)?
+        expire_in = sig.get_key_expires()
+        sig_generated = sig.when_generated()
+        if not uid_winner? or uid_winner.sig_generated < sig_generated
+            uid_winner = { expire_in, sig_generated }
+
+      if uid_winner?
```

Comment:
Yeah, I think it's possible.  The code should still work in that case, I believe.